pkg/lib-volume-populator/populator-machinery/controller.go

```diff
@@ -761,17 +773,14 @@ func (c *controller) syncPvc(ctx context.Context, key, pvcNamespace, pvcName str
 	// *** At this point the volume population is done and we're just cleaning up ***
 	c.recorder.Eventf(pvc, corev1.EventTypeNormal, reasonPodFinished, "Populator finished")
 
-	// If the pod still exists, delete it
-	if pod != nil {
-		err = c.kubeClient.CoreV1().Pods(populatorNamespace).Delete(ctx, pod.Name, metav1.DeleteOptions{})
+	// If PVC' still exists, delete it
+	if pvcPrime != nil {
+		err = c.kubeClient.CoreV1().PersistentVolumeClaims(populatorNamespace).Delete(ctx, pvcPrime.Name, metav1.DeleteOptions{})
 		if err != nil {
 			return err
 		}
-	}
 
-	// If PVC' still exists, delete it
-	if pvcPrime != nil {
-		err = c.kubeClient.CoreV1().PersistentVolumeClaims(populatorNamespace).Delete(ctx, pvcPrime.Name, metav1.DeleteOptions{})
+		err = c.ensureFinalizer(ctx, pvcPrime, pvcKuberenetesFinalizer, false)
```

Comment:
@bennyz @liranr23 did we reach a consensus on whether we want/need to remove the prime pvc?

Reply:
i think we did. both options are fine, since the status of the pvc prime is `Lost` and there is no intention to use it. i like the idea of deleting it without leaving it in terminating state. but if there is strong opinion to have it, that's fine as well.

Reply:
yes, that was the main concern - if we can identify when the PV switched back
but there was a case in which having the prime pvc can help us identifying the correlation between the populaor pod and the migration, @bennyz ?

Reply:
yes, I think we can put the prime PVCs to good use and find all the relevant pods, but I don't feel to strongly about this and I don't think we need to delay this PR for this optimization

Reply:
fair point

Reply:
yeah, I think we should to push this forward